hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/CustomAvroKeyGenerator.java

```diff
@@ -99,31 +102,42 @@ public static List<PartitionKeyType> getPartitionTypes(List<String> partitionPat
     if (partitionPathFields.size() == 1 && partitionPathFields.get(0).isEmpty()) {
       return Collections.emptyList(); // Corresponds to no partition case
     } else {
-      return partitionPathFields.stream().map(field -> {
-        Pair<String, CustomAvroKeyGenerator.PartitionKeyType> partitionAndType = getPartitionFieldAndKeyType(field);
-        return partitionAndType.getRight();
-      }).collect(Collectors.toList());
+      return partitionPathFields.stream().map(field -> getPartitionFieldAndKeyType(field).getRight())
+          .filter(Option::isPresent)
```

Comment:
We shouldn't have missing types right?. Should we be throwing an exception like above?

Reply:
Either all should be missing or none I think?

Reply:
This should already be checked during instantiation of custom keygen in `org.apache.hudi.keygen.CustomAvroKeyGenerator#getPartitionKeyGenerators`.

Reply:
I have made some changes to the APIs though so that it always reads the partition fields from table config.